### PR TITLE
[ADD] fieldservice_stage_server_action

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR adds a many2one field for fsm.stage to ir.server.action. The module also has an ir.server.action for each stage when an email should be sent out.

#142

### CODE REVIEW

File: fieldservice/models/fsm_order.py [hunk 1]
```diff
@@ -299,6 +299,41 @@ def _onchange_template_id(self):
...
+        email = ""
+        if self.person_id:
+            for person in self.person_ids:
+                if (person.email and (person != self.person_id)):
+                    email += person.email + ','
+        email = email[:-1]
+        self._compute_workers_not_selected = email
+
+    def _compute_outage_list(self):
+        rtype = self.env['res.partner.relation.type'].\
```
Comment (max3903) [inline fieldservice/models/fsm_order.py]: @osi-scampbell `fieldservice` does not depend on the module providing this model. It will not work.

**Comment by max3903** [comment]
@osi-scampbell Where do you make the relationship between a stage and its action?

File: fieldservice/models/fsm_order.py (hunk 1 above)
Comment (osi-scampbell) [reply fieldservice/models/fsm_order.py]: I see, I think it should be moved to pavlov_media_fieldservice then

**Comment by osi-scampbell** [comment]
> @osi-scampbell Where do you make the relationship between a stage and its action?

In fsm_stage.py I add action_id which is a link to ir.actions.server. When it is installed the user goes to a stage and sets the action there.

File: fieldservice/models/fsm_order.py (hunk 1 above)
Comment (max3903) [reply fieldservice/models/fsm_order.py]: Yes please.

Comment (osi-scampbell) [reply fieldservice/models/fsm_order.py]: Done

**Comment by max3903** [comment]
@osi-scampbell We should update the list of stages and set the action. See an example below.

in data/fsm_stage.xml:
```
<record id="fieldservice.fsm_stage_confirmed" model="fsm.stage">
    <field name="action_id" ref="fsm_order_parent_confirmed"/>
</record>
```

**Comment by osi-scampbell** [comment]
> @osi-scampbell We should update the list of stages and set the action. See an example below.
> 
> in data/fsm.stage.xml:
> 
> ```
> <record id="fieldservice.fsm_stage_confirmed" model="fsm.stage">
>     <field name="action_id" ref="fsm_order_parent_confirmed"/>
> </record>
> ```

So that manually sets the action_id so the user doesn't have to configure it themselves?

**Comment by max3903** [comment]
@osi-scampbell Yes!

**Comment by osi-scampbell** [comment]
Perfect, there are a few more spots I think that would be helpful in. I will add them now.

**Review by max3903 — APPROVED** [review]

File: fieldservice_stage_server_action/data/fsm_stage.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (patrickrwilson) [inline fieldservice_stage_server_action/data/fsm_stage.xml]: <?xml version="1.0" encoding="utf-8"?> not needed anymore in v12

File: fieldservice_stage_server_action/data/ir_server_action.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (patrickrwilson) [inline fieldservice_stage_server_action/data/ir_server_action.xml]: <?xml version="1.0" encoding="utf-8"?> not needed anymore in v12

File: fieldservice_stage_server_action/views/fsm_stage.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (patrickrwilson) [inline fieldservice_stage_server_action/views/fsm_stage.xml]: <?xml version="1.0" encoding="utf-8"?> not needed anymore in v12

File: fieldservice_stage_server_action/data/base_automation.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (patrickrwilson) [inline fieldservice_stage_server_action/data/base_automation.xml]: <?xml version="1.0" encoding="utf-8"?> not needed anymore in v12

**Review by patrickrwilson — APPROVED** [review]
Small changes

**Pull request merged**